### ISSUE TITLE
Unable to register in favorites

### ISSUE THREAD

**Issue by PiliJ** [BrowserWorks/waterfox#1106]
Hi
It is impossible for me to put a page of the forum of Promo2Day (https://www.promo2day.com) in the favorites (clicking on the star or via the Favorites menu).
The same problem does not occur with other sites, under Firefox or IE !
I don't know if it's related to the site or Waterfox...

**Comment by hawkeye116477** [comment]
For me it's working OK on Linux.

**Comment by Insamconcept** [comment]
For me too no problem (Linux)

**Comment by grahamperrin** [comment]
> … via the Favorites menu) …

https://www.promo2day.com/ bookmarked without difficulty using the **Bookmarks** menu in Waterfox Classic 56.2.13 build 20190820111231 on FreeBSD-CURRENT. 

Use the about:support page to verify your _Places_ database. There's a **Verify integrity** button.

**Comment by PiliJ** [comment]
Thanks for the tip grahamperrin ! There is indeed an integrity problem:
"> Task: checkIntegrity
- Unable to check database integrity
> Task: checkCoherence
- The task queue was cleared by an error in another task.
> Task: expire
- The task queue was cleared by an error in another task.
> Task: vacuum
- The task queue was cleared by an error in another task.
> Task: stats
- The task queue was cleared by an error in another task.
> Task: _refreshUI
- The task queue was cleared by an error in another task."

I'll try to solve it.

**Comment by grahamperrin** [comment]
Thanks. Google finds a few matches, this (from <https://support.mozilla.org/en-US/questions/1209416#answer-1088853>) might be useful: 

> … 
> 1. Close Firefox 
> 2. Go to profile and find places.sqlite file/files (when Firefox is open there are more than 1, but I did it when it was closed) 
> 3. Delete (or move to another folder) that file/files. 
> 4. Run Firefox, it should restore bookmarks from backup stored in bookmarkbackups folder in the profile (I suppose it did that). 
> 5. …

Loosely adapted, for Waterfox: 

1. about:profiles
2. open the root directory of the affected profile
3. quit Waterfox
4. if you see either of these files – `places.sqlite-shm` or `places.sqlite-wal` – wait for Waterfox to finish quitting
5. when the only visible `places.…` file is `places.sqlite`, trash (or set aside) `places.sqlite`

That should be enough for the next run of Waterfox to use the most recent automated backup.

**Comment by PiliJ** [comment]
Your solution recommended for Waterfox did not work. I guess the backup was corrupted too.
So I copied the profile of my laptop which was healthy and contains approximately the same data.
When Waterfox opens, an error message: "The bookmark and history system will not be operational."
So I followed the Firefox procedure that is here : https://support.mozilla.org/fr/kb/corriger-marque-pages-historique-pas-operationnel?redirectlocale=en-US&as=u&redirectslug=The+bookmarks+and+history+system+will+not+be+functional&utm_source=inproduct
Since nothing worked, I finally created a new profile and copied all the data from the old one, except places & favicons.sqlite… and it works fine now.
Thank you again !
